### PR TITLE
try to fix yum fail which crashes the whole job

### PR DESCRIPTION
**Pull request by amoskong**
The patchset added verify for scylla repo content, currently we don't know if it downloaded the right & valid scylla repo. And added many retry for yum commands to avoid network problem.

### CODE REVIEW

File: sdcm/utils.py [hunk 1]
```diff
@@ -140,3 +140,19 @@ def get_data_dir_path(*args):
 
 def get_job_name():
     return os.environ.get('JOB_NAME', 'local_run')
+
+
+def verify_scylla_repo(content, is_rhel_like=True):
```
Comment (bentsi) [inline sdcm/utils.py]: `verify_scylla_repo` -> `verify_scylla_repo_file`

File: sdcm/utils.py [hunk 2]
```diff
@@ -140,3 +140,19 @@ def get_data_dir_path(*args):
...
+    if is_rhel_like:
```
Comment (bentsi) [inline sdcm/utils.py]: add a message to log "Verifying Scylla repo file"

File: upgrade_test.py [hunk 3]
```diff
@@ -76,26 +76,19 @@ def upgrade_node(self, node):
...
             assert new_scylla_repo.startswith('http')
-            if node.is_rhel_like():
-                node.remoter.run('sudo curl -L {} -o /etc/yum.repos.d/scylla.repo'.format(new_scylla_repo))
-                node.remoter.run('sudo chown root:root /etc/yum.repos.d/scylla.repo')
-                node.remoter.run('sudo chmod 644 /etc/yum.repos.d/scylla.repo')
-                node.remoter.run('sudo yum clean all')
-            else:
-                node.remoter.run('sudo curl -L {} -o /etc/apt/sources.list.d/scylla.list'.format(new_scylla_repo))
-                node.remoter.run('sudo apt-get update')
+            node.download_scylla_repo(new_scylla_repo)
```
Comment (bentsi) [inline upgrade_test.py]: good :+1:

**Comment by amoskong** [comment]
@bentsi @roydahan 

http://jenkins.cloudius-systems.com:8080/job/scylla-3.0/job/3.0-test/job/scylla-rolling-upgrade-centos-test/base_version=2.3,label=gce3-qavpc,test_sst3=false,version_desc=3-0/197/display/redirect

This 3.0 rolling upgrade job failed for invalid repo or network problem, and another sub job (with sst3=true) is PASS. We need to backport this PR to branch-3.0

**Pull request merged**